pkg/refactor/eventstore/eventstore.go

```diff
@@ -0,0 +1,81 @@
+package eventstore
+
+import (
+	"context"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+)
+
+type CloudEventsStore interface {
+	Create(ctx context.Context,
+		nsID uuid.UUID,
+		fireTime time.Time,
+		event *cloudevents.Event,
+		eventID uuid.UUID,
```

Comment:
eventID is contained within event so it's not a necessary argument. Although I've also suggested that we pass in `[]byte` instead of a `cloudevents.Event`, so that should be fine.